pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -251,6 +248,14 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
           .put(MetadataKey.MIN_CONSUMING_FRESHNESS_TIME_MS.getName(), Long.toString(minConsumingFreshnessTimeMs));
     }
 
+    if (numSegmentsQueried > numSegmentsAcquired) {
+      String errorMessage =
+          String.format("Some segments could not be acquired: %d", numSegmentsQueried - numSegmentsAcquired);
```

Comment:
Collect the segment names unable to be acquired and put them into the error message

Reply:
Addressed.